### DISCUSSION TITLE
Minimal rootfs with graphics, sound & apt for chroot container, how-to?

### DISCUSSION THREAD

**Discussion by sunarowicz** [Joshua-Riek/ubuntu-rockchip#1108] [Q&A]
I would like to make a minimal rootfs with working graphics acceleration, sound and apt to be able to install anything needed. My intention is to use this rootfs for setting up the chroot container. As I understand I can make rootfs like this:

sudo ./build.sh --board=rock-5b --suite=noble --flavor=server -l -ro

But I expect to get still too large rootfs with many unnecessary stuff and have no idea if graphics and sound will be working in the server type flavor.

Can someone give me an idea how to do that, please?

Any help much appreciated.

**Comment by hakotani-o** [comment]
Hi sunarowicz.

Please modify scripts/build-rootfs.sh.

if [ "${PROJECT}" == "ubuntu" ]; then
    # Specific packages to install for ubuntu desktop
    (
        echo "ubuntu-desktop-rockchip"
        echo "oem-config-gtk"
        echo "ubiquity-frontend-gtk"
        echo "ubiquity-slideshow-ubuntu"
        echo "localechooser-data"
    ) >> config/package-lists/my.list.chroot
else
    # Specific packages to install for ubuntu server
    echo "THE PACKAGE NAME YOU WANT TO INSTALL" >> config/package-lists/my.list.chroot
fi

Thank you.

**Comment by sunarowicz** [comment]
Thank you for your interest, but this is not what I meant. If I understand that well, I can specify couple of Joshua's "meta" packages as ubuntu-desktop-rockchip or ubuntu-server-rockchip where you are pointing me at, but I expect these to install a lot of stuff I do not want to install. E.g. LibreOffice in case of ubuntu-desktop-rockchip. 

Maybe I should proceed from the opposite side and to debootstrap the minimal rootfs from http://archive.ubuntu.com/ubuntu, add Joshua's rockchip PPAs and then install what is needed to make accelerated graphics and sound to work. To be able run accelerated chromium browser with working sound for example from the chroot container. Would it work this way? What packages needs to be installed after adding PPAs if so?

**Reply by hakotani-o** [reply]
My understanding of English was wrong.
sorry.
If you don't need LibreOffice, use the terminal
sudo apt purge libreoffice-common libreoffice-style*
I think it will be deleted if you run.

 Do you want to delete unnecessary packages while running lb build?
 I'm not confident in writing the script, but it looks like this.

if [ "${PROJECT}" == "ubuntu" ]; then
    # Specific packages to install for ubuntu desktop
    (
        echo "ubuntu-desktop-rockchip"
        echo "oem-config-gtk"
        echo "ubiquity-frontend-gtk"
        echo "ubiquity-slideshow-ubuntu"
        echo "localechooser-data"
    ) >> config/package-lists/my.list.chroot
    echo '#!/bin/bash
             apt-get -y purge libreoffice-common libreoffice-style*' > config/hooks/200-my.purge.lists.chroot
else
    # Specific packages to install for ubuntu server
    echo "ubuntu-server-rockchip" >> config/package-lists/my.list.chroot
fi

Thank you.

**Comment by sunarowicz** [comment]
Hi, no problem at all, really. I mentioned LibreOffice just as an example. There is a ton of other packages that are not necessary for the minimalist rootfs. I definitely prefer to make as small as possible working rootfs and then add what is needed for the current purpose of chroot container over making the "fat" desktop rootfs and then removing what I sometime know and sometime just guess to be not necessary. This would be really ineffective and could lead into broken system.
